src/utils/cst_tokenstream.c

```diff
@@ -47,27 +48,193 @@ const cst_string *const cst_ts_default_postpunctuationsymbols =
 
 #define TS_BUFFER_SIZE 256
 
-static cst_string ts_getc(cst_tokenstream *ts);
-static cst_string internal_ts_getc(cst_tokenstream *ts);
+static void ts_getc(cst_tokenstream *ts);
+static void internal_ts_getc(cst_tokenstream *ts);
 
-static void set_charclass_table(cst_tokenstream *ts)
+
+int ts_charclass(const cst_string *const utf8char, int class,
+                 cst_tokenstream *ts)
 {
-    int i;
-    memset(ts->charclass, 0, 256);      /* zero everything */
-
-    for (i = 0; ts->p_whitespacesymbols[i]; i++)
-        ts->charclass[(unsigned char) ts->p_whitespacesymbols[i]] |=
-            TS_CHARCLASS_WHITESPACE;
-    for (i = 0; ts->p_singlecharsymbols[i]; i++)
-        ts->charclass[(unsigned char) ts->p_singlecharsymbols[i]] |=
-            TS_CHARCLASS_SINGLECHAR;
-    for (i = 0; ts->p_prepunctuationsymbols[i]; i++)
-        ts->charclass[(unsigned char) ts->p_prepunctuationsymbols[i]] |=
-            TS_CHARCLASS_PREPUNCT;
-    for (i = 0; ts->p_postpunctuationsymbols[i]; i++)
-        ts->charclass[(unsigned char) ts->p_postpunctuationsymbols[i]] |=
-            TS_CHARCLASS_POSTPUNCT;
-    return;
+    unsigned char *c = (unsigned char *) utf8char;
+    switch (cst_strlen(c))
```

Comment:
Maybe wrap this in an inline function like ```character_size()``` or similar to add slight documentation of what is happening.